src/Symfony/Component/Security/Http/Authentication/AuthenticationUtils.php

```diff
@@ -65,7 +65,13 @@ public function getLastAuthenticationError($clearSession = true)
      */
     public function getLastUsername()
     {
-        $session = $this->getRequest()->getSession();
+        $request = $this->getRequest();
+
+        if ($request->attributes->has(Security::LAST_USERNAME)) {
+            return $request->attributes->get(Security::LAST_USERNAME);
+        }
+
+        $session = $request->getSession();
 
         return null === $session ? '' : $session->get(Security::LAST_USERNAME);
```

Comment:
If $session have not contain "Security::LAST_USERNAME" in this case it should be null.